__tests__/app.test.js

```diff
@@ -37,82 +49,126 @@ test('200 - sends a list of topics to the client', () => {
         })
     })
 })
-describe('GET: 404 - bad request',() => {
-    test('404 - returns 404 if endpoint does not exist', () => {
-        return request(app)
-        .get('/api/giraffes')
-        .expect(404)
-        .then(({res}) => {
-            expect(res.statusMessage).toBe('Not Found')
-        })
-    })
-})
-describe('GET:/api/articles/:article_id', () => {
-    test('200 - responds with an article when given a valid id',() => {
-        return request(app)
-            .get('/api/articles/4')
-            .expect(200)
-            .then(({body}) => {
-                expect(body.article.title).toBe("Student SUES Mitch!")
-                expect(body.article.topic).toBe("mitch")
-                expect(body.article.author).toBe("rogersop")
-                expect(typeof body.article.created_at).toBe('string') 
-                expect(typeof body.article.article_img_url).toBe('string')
-            })
-    })
-    test('404 - responds with 404 "article does not exist" when given a valid id that is not present',() => {
-        return request(app)
-            .get('/api/articles/999')
-            .expect(404)
-            .then((response) => {
-                expect(response.body.msg).toBe('article does not exist');
-            })
-    })
-    test('400 - responds with 400 bad request when given a non-valid id',() => {
-        return request(app)
-            .get('/api/articles/scooby')
-            .expect(400)
-            .then(({body}) => {
-                expect(body.msg).toBe('Invalid id type')
-            })
-    })
-})
+
 describe('GET:/api/articles', () => {
-    test('200 - responds with an array of all articles',() => {
-        return request(app)
-            .get('/api/articles')
-            .expect(200)
-            .then(({body}) => {
-                body.articles.forEach(article => {
-                    expect(article).toMatchObject({
-                    author: expect.any(String),
-                    title: expect.any(String),
-                    article_id: expect.any(Number),
-                    topic: expect.any(String),
-                    created_at: expect.any(String), 
-                    votes: expect.any(Number),
-                    article_img_url: expect.any(String),
-                    comment_count: expect.any(String)
+    describe('GET:/api/articles', () => {
+        test('200 - responds with an array of all articles',() => {
+            return request(app)
+                .get('/api/articles')
+                .expect(200)
+                .then(({body}) => {
+                    body.articles.forEach(article => {
```

Comment:
I would add a test for the length of the array before using a forEach to test the properties of each array. This way if you are receiving an empty array your tests inside the forEach would be skipped but the length test would still run.